### PR TITLE
chore: Switch from `eth2_ssz` to `ethereum_ssz` crate

### PR DESCRIPTION
**Pull request by morph-dev**
### What was wrong?
We depend on `ethereum_types v.0.12.1` and we can't upgrade it to the newer versions, because they are no longer compatible with `eth2_ssz` library that we use. Solution is to switch to `ethereum_ssz` and upgrade dependencies together.

See #882 for details

### How was it fixed?
Some crates were replaced, while some were upgraded to newer version:

`eth2_ssz` v.0.4.0 -> `ethereum_ssz` v.0.5.3
`eth2_ssz_derive` v.0.3.0 -> `ethereum_ssz_derive` v.0.5.3
`eth2_ssz_types` v.0.2.1 -> `ssz_types` v.0.5.4

`ethereum-types`: v.0.12.1 -> v.0.14.1
`tree_hash`: v.0.4.0 -> v.0.5.2
`tree_hash_derive`: v.0.4.0 -> v.0.5.2
`keccak-hash`: v.0.8.0 -> v.0.10.0

Bonus: Some files were no longer needed: `ethportal-api/src/types/wrapped/h160.rs`

### To-Do

- [x] Wait until `eth_trie` is upgraded and released, and switch to new version as well. PR created: [carver/eth-trie.rs/#32](https://github.com/carver/eth-trie.rs/pull/32).
- [x] Clean up commit history and use [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/).

### CODE REVIEW

File: ethportal-api/src/types/execution/block_body.rs [hunk 1]
```diff
@@ -138,6 +138,7 @@ impl BlockBody {
         }
 
         trie.root_hash()
+            .map(|root| H256::from_slice(root.as_bytes()))
```
Comment (carver) [inline ethportal-api/src/types/execution/block_body.rs:141]: Both are ethereum_types::H256, can't we skip the mapping?

**Review by carver — APPROVED** [review]
Looks good! Yay for upgrading dependencies and removing code.

**Comment by carver** [comment]
eth-trie v0.4 is published

File: ethportal-api/src/types/execution/block_body.rs (hunk 1 above)
Comment (morph-dev) [reply ethportal-api/src/types/execution/block_body.rs]: As discussed, it was needed before `eth_trie` was upgraded and new version published. No longer needed.

**Comment by morph-dev** [comment]
Upgraded eth_trie to v0.4. Ready to be merged.

**Review by KolbyML — APPROVED** [review]
:shipit:

**Pull request merged**